compiler/src/yul/mappers/expressions.rs

```diff
@@ -215,310 +129,206 @@ pub fn spanned_expression<'a>(span: &Span, exp: &fe::Expr<'a>) -> Spanned<fe::Ex
-        Some(ContractDef::Map { index, map }) => Ok(ExtExpression {
-            expression: literal_expression! {(index)},
-            location: Location::Storage,
-            typ: Type::Map(map),
-        }),
-        Some(ContractDef::Function { .. }) => unimplemented!(),
-        Some(ContractDef::Event(_)) => {
-            Err(CompileError::static_str("invalid use of event definition"))
-        }
-        None => Err(CompileError::static_str("unknown contract definition")),
+    context: &Context,
+    exp: &Spanned<fe::Expr>,
+) -> Result<yul::Expression, CompileError> {
+    match context.get_expression(exp) {
+        Some(ExpressionAttributes {
+            typ: Type::Map(_),
+            location: Location::Storage { index },
+        }) => Ok(literal_expression! { (index) }),
+        _ => unimplemented!(),
     }
 }
 
 #[cfg(test)]
 mod tests {
     use crate::yul::mappers::expressions::{
         expr,
-        ExtExpression,
         Location,
     };
-    use crate::yul::namespace::scopes::{
-        ContractScope,
-        FunctionScope,
-        ModuleScope,
-        Shared,
-    };
-    use crate::yul::namespace::types::{
+    use fe_parser as parser;
+    use fe_semantics::namespace::types::{
         Array,
         Base,
         FixedSize,
         Map,
         Type,
     };
-    use fe_parser as parser;
+    use fe_semantics::test_utils::ContextHarness;
+    use fe_semantics::{
+        Context,
+        ExpressionAttributes,
+    };
     use rstest::rstest;
-    use std::rc::Rc;
-
-    fn scope() -> Shared<FunctionScope> {
-        let module_scope = ModuleScope::new();
-        let contract_scope = ContractScope::new(module_scope);
-        FunctionScope::new(contract_scope)
-    }
 
-    fn map(scope: Shared<FunctionScope>, src: &str) -> ExtExpression {
+    fn map(context: &Context, src: &str) -> String {
         let tokens = parser::get_parse_tokens(src).expect("Couldn't parse expression");
         let expression = &parser::parsers::expr(&tokens[..])
             .expect("Couldn't build expression AST")
             .1;
 
-        expr(scope, expression).expect("Couldn't map expression AST")
+        expr(context, expression)
+            .expect("Couldn't map expression AST")
+            .to_string()
     }
 
     #[test]
     fn map_sload_u256() {
-        let scope = scope();
-        scope.borrow_mut().contract_scope().borrow_mut().add_map(
-            "foo".to_string(),
-            Map {
-                key: FixedSize::Base(Base::Address),
-                value: FixedSize::Base(Base::U256),
-            },
-        );
-
-        let result = map(scope, "self.foo[3]");
-
-        assert_eq!(
-            result.expression.to_string(),
-            "sloadn(dualkeccak256(0, 3), 32)"
-        );
-        assert_eq!(result.location, Location::Value);
-        assert_eq!(result.typ, Type::Base(Base::U256));
-    }
-
-    #[test]
-    fn map_sload_array_and_address() {
```

Comment:
Was this test dropped without replacement?

Reply:
Yeah, it looks like this test was originally added to make sure we can map expressions with multiple storage values in scope. It didn't really seem necessary to keep this anymore since things are more straight forward with context. It would basically test the other two map tests in the same test, which seems a bit redundant.